client/ladder/action-creators.ts

```diff
@@ -17,37 +18,46 @@ export function getRankings(
   spec: RequestHandlingSpec<void>,
 ): ThunkAction {
   return abortableThunk(spec, async dispatch => {
-    const fetchTime = new Date()
-    const lastFetchTime = lastFetchTimeByMatchmakingType.get(matchmakingType) ?? new Date(0)
+    // `performance.now()` can return value that's less than our cache time, so we floor it to that.
+    const fetchTime = performance.now() + LADDER_RANKINGS_CACHE_TIME_MS
+    const lastFetchTime = lastFetchTimeByMatchmakingType.get(matchmakingType) ?? 0
+    const lastSearchTime = lastFetchTimeByMatchmakingType.get(matchmakingType) ?? 0
     const lastSearchQuery = lastSearchQueryByMatchmakingType.get(matchmakingType) ?? ''
 
-    if (
-      Number(fetchTime) - Number(lastFetchTime) < LADDER_RANKINGS_CACHE_TIME_MS &&
-      lastSearchQuery === searchQuery
-    ) {
-      return
+    if (searchQuery) {
```

Comment:
`searchQuery` isn't specified as being possibly `undefined`, so this feels incredibly weird to me. imo you should just split this into 2 functions: one with a searchQuery and one without, since they are almost entirely separate code paths anyway.

They should probably also be separate redux actions

Reply:
Yeah, I'm basically relying here on an empty string being a falsy value, but I wasn't really happy with that myself. I also debated with myself about separating this into 2 functions and redux actions, which in hindsight I should've done ^^